packages/admin/admin/src/form/FinalFormNumberInput.tsx

```diff
@@ -0,0 +1,84 @@
+import { InputBase, InputBaseProps } from "@mui/material";
+import * as React from "react";
+import { FieldRenderProps } from "react-final-form";
+import { useIntl } from "react-intl";
+
+import { ClearInputAdornment } from "../common/ClearInputAdornment";
+
+export type FinalFormNumberInputProps = InputBaseProps &
+    FieldRenderProps<number | string> & {
```

Comment:
Should this not always be a number? Why is string needed? 🤔 

```suggestion
    FieldRenderProps<number> & {
```

Reply:
We still use strings for the input at some points, e.g. in the getFormattedValue function

`    const getFormattedValue = React.useCallback(
        (value, decimals: number) => {
            const formattedValue =
                value !== undefined ? intl.formatNumber(value, { minimumFractionDigits: decimals, maximumFractionDigits: decimals }) : "";
            return formattedValue;
        },
        [intl],
    );`

and later also in the useEffect:

`  React.useEffect(() => {
        if (input.value !== "") {
            setFormattedNumberValue(getFormattedValue(input.value, decimals));
        } else {
            setFormattedNumberValue("");
        }
    }, [getFormattedValue, decimals, input.value]);`

Reply:
the value type of this field is always number, that's the whole point of it.

(yes, Internally it uses strings - but that must not be visible from the outside)